server/src/controller/userController.js

```diff
@@ -0,0 +1,27 @@
+import User from "../models/userModel.js";
+
+//TODO: improve code below with validations
+const userLogin = async (req, res) => {
+  res.json({ userLoginTest: "yeah" });
+};
+
+const userSignUp = async (req, res) => {
+  const name = req.body.name;
+  const email = req.body.email;
+  const postalCode = req.body.postalCode;
+  const password = req.body.password;
+
+  try {
+    const user = await User.create({
+      name,
+      email,
+      postalCode,
+      password,
```

Comment:
We prob want to hash the users password and not want to store the exact string. 
I think possible Argon2 is whats good to use right now
https://www.npmjs.com/package/argon2

Reply:
yeah, agree. we will hash it with argon2 or bcrypt, it will be encrypted together with salt.  

i think we can work on encryption when we work on user auth, which is another issue i have created. just tried to reduce the size of prs and focus on one at a time. but that's really good point and it's so critical we shouldn't miss.

how does the database structure look like to you guys, i tried but still i feel like maybe there should be more efficient way to do it? 🤔 . 

we will probably change it anyway while we progress but i would like to try to make it as close as possible.

let me know if you guys have any questions or suggestions while examine this.

Reply:
also need to do some research about which one is better to choose, argon2 or Bcrypt.

Reply:
> We should also consider generating a random unique id string to each user and use that to identify them throughout the application as many apps do, as that will allow us to deal with situations in the future where it may be necessary to change a user's name, email or other information.

Mongo DB by default adds a unique id